exporter/exporterhelper/internal/bounded_memory_queue.go

```diff
@@ -17,85 +17,144 @@ import (
 // where the queue is bounded and if it fills up due to slow consumers, the new items written by
 // the producer are dropped.
 type boundedMemoryQueue struct {
-	stopWG       sync.WaitGroup
-	size         *atomic.Uint32
-	stopped      *atomic.Bool
+	size         int
+	eventChan    chan any
 	items        chan Request
-	capacity     uint32
+	capacity     int
 	numConsumers int
+	stopped      *atomic.Bool
+	overflow     *atomic.Bool
 }
 
 // NewBoundedMemoryQueue constructs the new queue of specified capacity, and with an optional
 // callback for dropped items (e.g. useful to emit metrics).
 func NewBoundedMemoryQueue(capacity int, numConsumers int) ProducerConsumerQueue {
 	return &boundedMemoryQueue{
 		items:        make(chan Request, capacity),
-		stopped:      &atomic.Bool{},
-		size:         &atomic.Uint32{},
-		capacity:     uint32(capacity),
+		eventChan:    make(chan any),
```

Comment:
What is the `eventChan` used for?  I wouldn't recommend using `any` since readers would need to check or infer type which could lead for a bad time.

Reply:
it's the internal event loop channel of the queue.